src/System.Numerics.Vectors/src/System/Numerics/Matrix4x4.cs

```diff
@@ -2149,12 +2300,23 @@ public static Matrix4x4 Multiply(Matrix4x4 value1, float value2)
         /// <param name="value1">The first matrix to compare.</param>
         /// <param name="value2">The second matrix to compare.</param>
         /// <returns>True if the given matrices are not equal; False if they are equal.</returns>
-        public static bool operator !=(Matrix4x4 value1, Matrix4x4 value2)
+        public static unsafe bool operator !=(Matrix4x4 value1, Matrix4x4 value2)
         {
-            return (value1.M11 != value2.M11 || value1.M12 != value2.M12 || value1.M13 != value2.M13 || value1.M14 != value2.M14 ||
-                    value1.M21 != value2.M21 || value1.M22 != value2.M22 || value1.M23 != value2.M23 || value1.M24 != value2.M24 ||
-                    value1.M31 != value2.M31 || value1.M32 != value2.M32 || value1.M33 != value2.M33 || value1.M34 != value2.M34 ||
-                    value1.M41 != value2.M41 || value1.M42 != value2.M42 || value1.M43 != value2.M43 || value1.M44 != value2.M44);
+            if (Sse.IsSupported)
+            {
+                return
+                    Sse.MoveMask(Sse.CompareEqual(Sse.LoadVector128(&value1.M11), Sse.LoadVector128(&value2.M11))) != 0xF ||
```

Comment:
`MoveMask(CompareNotEqual()) != 0` is probably more efficient

Reply:
just tested, you are right, it's 30% more efficient. Also, I tried to add some fast-out paths (compare first field or first row as simple m1.M11 == m2.M11 && ...) but it gave a minor improvement for some cases and major regression in anothers.